### PR TITLE
Hw jdbc

### PR DESCRIPTION
**Pull request by Martynice**

### CODE REVIEW

File: src/main/resources/init_db.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CREATE SCHEMA `internet-shop` DEFAULT CHARACTER SET utf8 ;
+CREATE TABLE `internet-shop`.`products` (
+  `product_id` BIGINT(11) NOT NULL AUTO_INCREMENT,
+  `name` VARCHAR(255) NOT NULL,
+  `price` DOUBLE NOT NULL,
+  `deleted` TINYINT NOT NULL DEFAULT 0,
+  PRIMARY KEY (`product_id`));
+
```
Comment (frostpv) [inline src/main/resources/init_db.sql]: пустая строка

**Review by frostpv — APPROVED** [review]

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            }
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create product " + product, e);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT * FROM products WHERE product_id = ?";
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: let's get only existing products

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Long productId = resultSet.getLong("productId");
+                String name = resultSet.getString("name");
+                double price = resultSet.getDouble("price");
+                Product product = new Product(name, price);
+                product.setId(productId);
+                return Optional.of(product);
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_21.html
first point

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get product", e);
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: let's add more information about product

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products";
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: let's get only existing products

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all products", e);
+        }
+        return products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE products SET name = ?, price = ? WHERE product_id = ?";
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: can we update deleted data?

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 2 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 3 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: added

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 4 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: added

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 5 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 6 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**